cedar-policy-validator/src/cedar_schema/grammar.lalrpop

```diff
@@ -82,21 +85,35 @@ match {
 
     // other tokens
     ",", ";", ":", "::", "{", "}", "[", "]",
-    "<", ">", "=", "?",
+    "<", ">", "=", "?", "@", "(", ")",
 
 }
 
+#[inline]
+AnyIdent: Node<AnyId> = {
+    <l:@L> <id:IDENTIFIER> <r:@R>
+        => Node::with_source_loc(AnyId::new_unchecked(id), Loc::new(l..r, Arc::clone(src))),
+}
+
+// Annotations := {'@' IDENTIFIER '(' String ')'}
+Annotation: Node<Annotation> = {
+    <l:@L> "@" <key:AnyIdent> <value: ("(" <STR> ")")> <r:@R> => Node::with_source_loc((key, value), Loc::new(l..r, Arc::clone(src)))
+}
+
+Annotated<E>: Annotated<E> = {
+   <annotations: Annotation*> <e:E> => Annotated {data: e, annotations},
+}
 
 // Schema := {Namespace}
 pub Schema: ASchema = {
     <ns: Namespace*> => ns,
 }
 
 // Namespace := 'namespace' Path '{' {Decl} '}'
-Namespace: Node<Namespace> = {
-    <l:@L> NAMESPACE <p: Path> "{" <decls: Decl*> "}" <r:@R>
-        => Node::with_source_loc(Namespace { name: Some(Node::with_source_loc(p, Loc::new(l..r, Arc::clone(src)))), decls}, Loc::new(l..r, Arc::clone(src))),
-     <l:@L> <decl: Decl> <r:@R> => Node::with_source_loc(Namespace {name: None, decls: vec![decl]}, Loc::new(l..r, Arc::clone(src))),
+Namespace: Annotated<Namespace> = {
+    <annotations: Annotation*> NAMESPACE <p: Path> "{" <decls: Annotated<Decl>*> "}"
```

Comment:
What's the `Annotated<E>` production doing if you still need to write out ` <annotations: Annotation*>` here?

Reply:
lalrpop errors if I use `Annotated<E>` here, i.e., the grammar is problematic.